src/components/ManaCalculator/components/ManaCalculatorInput.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { JSX, useEffect, useState } from 'react';
+
+interface IManaCalculatorCustomInput {
+  value: string | number;
+  onChange: (value: string) => void;
+  className?: string;
+  min?: number;
+  max?: number;
+}
+
+export function ManaCalculatorInput({
+  value,
+  onChange,
+  className,
+  min,
+  max,
+}: IManaCalculatorCustomInput): JSX.Element {
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [inputValue, setInputValue] = useState<string | number>(value);
+
+  useEffect(() => {
+    try {
+      verifyMinAndMax(value);
+      setErrorMessage('');
+    } catch (error) {
+      setErrorMessage(error.message);
+    }
+  }, [min, max]);
+
+  function verifyMinAndMax(value: string | number) {
+    let message = '';
+
+    if (!isNaN(min) && Number(value) < min) {
+      message = 'Minimum value is ' + min;
+    }
+    if (!isNaN(min) && Number(value) > max) {
+      message = 'Maximum value is ' + max;
+    }
+
+    if (
+      !isNaN(min) &&
+      !isNaN(max) &&
+      (Number(value) < min || Number(value) > max)
+    ) {
+      message = 'Min: ' + min + ', Max: ' + max;
+    }
+    if (message) {
+      throw new Error(message);
+    }
+  }
+
+  function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
+    try {
+      setInputValue(e.target.value);
+      verifyMinAndMax(e.target.value);
+      setErrorMessage('');
+      onChange(e.target.value);
+    } catch (error) {
+      setErrorMessage(error.message);
+    }
+  }
+
+  return (
+    <div
+      className={`mana_calculator__custom_wrapper ${
+        errorMessage ? 'errored' : ''
+      }`}
+    >
+      <input
+        className={className}
+        min={min}
+        max={max}
+        value={inputValue}
+        onChange={handleInputChange}
+      />
+      {errorMessage && <p className='mana_calculator__error'>{errorMessage}</p>}
+    </div>
+  );
+}
```

Comment:
```suggestion
function verifyMinAndMax(value: string | number, min: number, max: number) {
  let message = '';

  if (!isNaN(min) && Number(value) < min) {
    message = 'Minimum value is ' + min;
  }
  if (!isNaN(min) && Number(value) > max) {
    message = 'Maximum value is ' + max;
  }

  if (
    !isNaN(min) &&
    !isNaN(max) &&
    (Number(value) < min || Number(value) > max)
  ) {
    message = 'Min: ' + min + ', Max: ' + max;
  }
  
  return message
}

interface IManaCalculatorCustomInput {
  value: string | number;
  onChange: (value: string) => void;
  className?: string;
  min?: number;
  max?: number;
}

export function ManaCalculatorInput({
  value,
  onChange,
  className,
  min,
  max,
}: IManaCalculatorCustomInput): JSX.Element {
  const [errorMessage, setErrorMessage] = useState('');

  function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
    const error =  verifyMinAndMax(e.target.value);
    if(error){
       setErrorMessage(error);
    } else {
       onChange(e.target.value);
    }
  }

  return (
    <div
      className={`mana_calculator__custom_wrapper ${
        errorMessage ? 'errored' : ''
      }`}
    >
      <input
        className={className}
        min={min}
        max={max}
        value={value}
        onChange={handleInputChange}
      />
      {errorMessage && <p className='mana_calculator__error'>{errorMessage}</p>}
    </div>
  );
}
```

Reply:
With this changes the problems of @Dr-Electron should be gone

warning: I wrote this from GitHub, so there might some typos fyi

Reply:
Hey @marc2332 , the problems dr electron says he has are because the maxSupply are calculated differently than the one he made previously it seems. The max and min values work fine

Reply:
Also, if doing an `if else` instead of `try catching` the error would make it harder to get the error when validating the change. This way you can throw the error in the onChange and receive it in the input. Other way would require passing an error message externally as prop